### PR TITLE
rename encryptedCheckInterval and encryptedCheckTimeout and increase …

### LINKED ISSUE #77741: EBS API polling when creating a volume
**What happened**:

When creating an EBS volume, we're currently waiting for the volume be in the correct state before proceeding:

https://github.com/kubernetes/kubernetes/blob/88ed776c6b447b1882a5fe99df0cfcaba1da4e72/staging/src/k8s.io/legacy-cloud-providers/aws/aws.go#L2398-L2421

However, we're using the interval defined in `encryptedCheckInterval` (e.g., 1 second), which was meant to be used for another reason and might be too soon for this specific case. Also, this might contribute to the API quota being exceeded.

CC @gnufied @leakingtapan 
/sig storage
/sig aws

### PR DESCRIPTION
**Pull request by rusik69**
…encryptedCheckInterval

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
/kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
rename encryptedCheckInterval and encryptedCheckTimeout and increase encryptedCheckInterval to 5s
**Which issue(s) this PR fixes**:

Fixes #77741 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
NONE

```release-note
rename encryptedCheckInterval and encryptedCheckTimeout and increase encryptedCheckInterval to 5s
```

### CODE REVIEW

File: staging/src/k8s.io/legacy-cloud-providers/aws/aws.go [hunk 1]
```diff
@@ -235,12 +235,12 @@ const (
...
 	createTagSteps        = 9
 
-	// encryptedCheck* is configuration of poll for created volume to check
+	// createEBSCheck* is configuration of poll for created volume to check
 	// it has not been silently removed by AWS.
 	// On a random AWS account (shared among several developers) it took 4s on
 	// average.
-	encryptedCheckInterval = 1 * time.Second
-	encryptedCheckTimeout  = 30 * time.Second
+	createEBSCheckInterval = 5 * time.Second
```
Comment (leakingtapan) [inline staging/src/k8s.io/legacy-cloud-providers/aws/aws.go]: Let's keep it the same as EBS CSI driver: https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/pkg/cloud/cloud.go#L742

**Review by leakingtapan — COMMENTED** [review]
Thx for adding the fix.

File: staging/src/k8s.io/legacy-cloud-providers/aws/aws.go [hunk 2]
```diff
@@ -2418,7 +2418,7 @@ func (c *Cloud) waitUntilVolumeAvailable(volumeName KubernetesVolumeID) error {
 		return err
 	}
 
-	err = wait.Poll(encryptedCheckInterval, encryptedCheckTimeout, func() (done bool, err error) {
+	err = wait.Poll(createEBSCheckInterval, createEBSCheckTimeout, func() (done bool, err error) {
```
Comment (gnufied) [inline staging/src/k8s.io/legacy-cloud-providers/aws/aws.go]: I expected this to use exponential backoff and not a flat polling.

Comment (leakingtapan) [reply staging/src/k8s.io/legacy-cloud-providers/aws/aws.go]: Flat polling works pretty well so far. This change should focus on fixing the regression instead of adding new change. WDYT

Comment (gnufied) [reply staging/src/k8s.io/legacy-cloud-providers/aws/aws.go]: If upper limit as proposed in this PR is 30 seconds, we are probably fine. But, I am curious if 30s is indeed enough as upper bound of time limit for volume to become available. 

Even with 30s - we will currently poll approx. 10 times when checking for volume to become available. On average at least 2-3 times but if we used exponential backoff we will just check 5 times. From my experience - these numbers tend to add up and cause a cluster's API usage to go up. Since we have been trying to reduce AWS API usage as much as possible, my vote is to increase the max. wait time to higher value (may be 2mins. In recent benchmarks @jsafrane did he found 8s to be upper bound for 250 vols. he created) and use exponential backoff. 

We could also may be talk to EBS team about expected latency when volume becomes available after creation. Not waiting properly in `CreateVolume` call will cause created volume to become orphan.

Comment (gnufied) [reply staging/src/k8s.io/legacy-cloud-providers/aws/aws.go]: Please note - issues around this may have been masked until now because afaik most clusters use unencrypted volumes and hence we may not have sufficient evidence to prove that 30s time limit worked.

Comment (loqutus) [reply staging/src/k8s.io/legacy-cloud-providers/aws/aws.go]: committed exponential backoff for 2 minutes

**Comment by jsafrane** [comment]
@loqutus, please squash the commits, we don't need 4 commits for such simple fix.

And please edit the release note - it's for users and they are not interested in renaming of some internal functions, mentioning switch to exponential backoff is IMO fine.

The code itself tools IMO fine, @leakingtapan @gnufied WDYT?

**Comment by gnufied** [comment]
I agree and also we should (IMO) cherry-pick this commit to 1.14 because without it, we may notice some API quota related regressions in large clusters.

**Comment by jsafrane** [comment]
@loqutus, would you mind rebasing this PR and removing the first commit (it's unrelated to this PR)?

File: test/e2e/apimachinery/crd_watch.go [hunk 3]
```diff
@@ -80,35 +79,35 @@ var _ = SIGDescribe("CustomResourceDefinition Watch", func() {
 			noxuResourceClient := newNamespacedCustomResourceClient(ns, f.DynamicClient, noxuDefinition)
 
 			watchA, err := watchCRWithName(noxuResourceClient, watchCRNameA)
-			gomega.Expect(err).NotTo(gomega.HaveOccurred(), "failed to watch custom resource: %s", watchCRNameA)
+			framework.ExpectNoError(err, "failed to watch custom resource: %s", watchCRNameA)
```
Comment (justinsb) [inline test/e2e/apimachinery/crd_watch.go]: The gomega Expect(err).NotTo(HaveOccurred) removal is good, but should be a separate PR (please)

**Comment by justinsb** [comment]
Thanks for fixing this @loqutus - I don't see exponential backoff in this PR - did you forget to push ?

I believe this is hitting us because it was changed to check all volumes (not just KMS-encrypted volumes) in ac6d32a65a2bf418a4c55fc5f979fd5fb89a2141

**Comment by gnufied** [comment]
@justinsb this PR def. had exponential backoff when I last looked but now it appears to be gone. I would think a rebase gone wrong?

**Comment by loqutus** [comment]
sorry, rebase has gone wrong indeed. i've created a new pr [https://github.com/kubernetes/kubernetes/pull/78601](url)
